src/Cardano/Shell/NodeIPC/ServerExample.hs

```diff
@@ -118,11 +124,14 @@ receieveMessages serverReadHandle = do
     reply   <- readServerMessage -- Reply
     return (started, reply)
 
-getHandleFromEnv :: String -> IO Handle
+getHandleFromEnv :: String -> IO (Either NodeIPCException Handle)
```

Comment:
Does `NodeIPCException` have an `Exception` instance? If so, it `Exception` instance should be dropped and the type renamed to something like `NodeIPCError`

The idea is that if an error type `e` is used as `Either e`, is should not be possible to throw it as an exception.

Reply:
If `NodeIPCException` does not have an `Exception` instance (or if it us removed), it should should be renamed to something like `NodeIPCError`

Reply:
Ah okay